[EDU-1658] - Chat room reactions examples (#2287)
ably/docs: examples/chat-room-reactions/page.md

## Resources

// React brief (Only visible if viewing the React example)

Use the following components to add room reactions into a chat application:

* [ChatClientProvider](https://ably.com/docs/chat/setup?lang=react#instantiate): initializes and manages a shared chat client instance, passing it down through React context to enable real-time chat functionality across the application.
* [ChatRoomProvider](https://ably.com/docs/chat/rooms?lang=react#create): manages the state and functionality of a specific chat room, providing access to messages, participants, and real-time interactions within that room via React context.
* [useRoom()](https://ably.com/docs/chat/rooms?lang=react#create) hook: a hook to manage the state and interaction for a chat “room”, allowing users to join, send messages, listen for messages, and use the other chat functions such as typing indicators.

> * [useRoomReactions()](https://ably.com/docs/chat/rooms/reactions?lang=react#subscribe) hook: a hook to manage the reactions sent by users within a chat room.

Reviewer: Should these be code styled like the methods are?

# Room reactions for chat applications

Use room reactions to enable users to express their feelings within a chat application.

Room reactions enable you to render when another chat user send a reaction to the room. They are most commonly used to display emojis such as a thumbs up (👍) or a heart (❤️).

> Room reactions can add value to many different chat applications. One example is in 1:1 customer support applications, where agents and clients can express their emotions or feedback instantly using reactions. In larger chat applications, such as a live streaming application, room reactions can enhance user engagement by allowing viewers to react to content in real-time.

Reviewer: ```suggestion
Room reactions add value to different applications. They enable users to quickly express their feelings about what is happening on screen. One such example is in a live streaming application where users can use the 👍 emoji if they're enjoying the content, and watch others react in realtime too.
```

> Room reactions can be implemented using [Ably Chat](https://ably.com/docs/products/chat). The Chat SDK contains a set of purpose-built APIs that abstract away the complexities involved in architecting chat features. It is built on top of Ably's core platform, and so it provides the same performance guarantees and scaling potential.

Reviewer: ```suggestion
Room reactions are implemented using [Ably Chat](https://ably.com/docs/products/chat). The Chat SDK contains a set of purpose-built APIs that abstract away the complexities involved in architecting chat features. It is built on top of Ably's core platform, and so it provides the same performance guarantees and scaling potential.
```

## Resources

// React brief (Only visible if viewing the React example)

Use the following components to add room reactions into a chat application:

* [ChatClientProvider](https://ably.com/docs/chat/setup?lang=react#instantiate): initializes and manages a shared chat client instance, passing it down through React context to enable real-time chat functionality across the application.
* [ChatRoomProvider](https://ably.com/docs/chat/rooms?lang=react#create): manages the state and functionality of a specific chat room, providing access to messages, participants, and real-time interactions within that room via React context.
* [useRoom()](https://ably.com/docs/chat/rooms?lang=react#create) hook: a hook to manage the state and interaction for a chat “room”, allowing users to join, send messages, listen for messages, and use the other chat functions such as typing indicators.

> * [useRoomReactions()](https://ably.com/docs/chat/rooms/reactions?lang=react#subscribe) hook: a hook to manage and track the reactions sent by users within a chat room.

Reviewer: ```suggestion
* [useRoomReactions()](https://ably.com/docs/chat/rooms/reactions?lang=react#subscribe) hook: a hook to manage the reactions sent by users within a chat room.
```

(careful here, as tracking suggests something like aggregation which isn't supported yet)
